src/std/fs.ab

```diff
@@ -108,3 +108,34 @@ pub fun glob_multiple(paths: [Text]): [Text]? {
 pub fun glob(path: Text): [Text]? {
     return glob_multiple([path])?
 }
+
+/// Extract the file detecting from the filename the extension
+/// Supports: bz2, gz, xz, bz2, deb, rar, rpm, tar(gz/xz/bz), zip(war/jar), 7z
+/// Note: Not all the commands supports the output folder path
+pub fun extract(path: Text, target: Text): Bool{
+    if file_exist(path) {
+        if {
+            contains_multiple(path, [".tar.bz2", ".tbz", ".tbz2"]): trust $tar xvjf "{path}" -C "{target}"$
+            contains_multiple(path, [".tar.gz", ".tgz"]): trust $tar xzf "{path}" -C "{target}"$
+            contains_multiple(path, [".tar.xz", ".txz"]): trust $tar xJf "{path}" -C "{target}"$
+            contains_multiple(path, [".bz2"]): trust $bunzip2 "{path}"$
+            contains_multiple(path, [".deb"]): trust $dpkg-deb -xv "{path}" "{target}"$
+            contains_multiple(path, [".gz"]): trust $gunzip "{path}"$
```

Comment:
Does anyone know if `gzip -d` is more canonical than `gunzip`?

Reply:
I see that is another binary